core/corehttp/gateway_handler_tar.go

```diff
@@ -0,0 +1,65 @@
+package corehttp
+
+import (
+	"context"
+	"html"
+	"net/http"
+	"time"
+
+	files "github.com/ipfs/go-ipfs-files"
+	"github.com/ipfs/go-ipfs/tracing"
+	ipath "github.com/ipfs/interface-go-ipfs-core/path"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+	"go.uber.org/zap"
+)
+
+func (i *gatewayHandler) serveTAR(ctx context.Context, w http.ResponseWriter, r *http.Request, resolvedPath ipath.Resolved, contentPath ipath.Path, begin time.Time, logger *zap.SugaredLogger) {
+	ctx, span := tracing.Span(ctx, "Gateway", "ServeTAR", trace.WithAttributes(attribute.String("path", resolvedPath.String())))
+	defer span.End()
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// // Set Cache-Control and read optional Last-Modified time
+	// modtime := addCacheControlHeaders(w, r, contentPath, resolvedPath.Cid())
```

Comment:
Below should be ok, but for it to set correct Etag, we need update the way we create Etag suffix in: 

https://github.com/ipfs/go-ipfs/blob/1a3b41c23be70de3ea9273c283845025bbb59b44/core/corehttp/gateway_handler.go#L876-L877

```suggestion
	// Set Cache-Control and read optional Last-Modified time
	modtime := addCacheControlHeaders(w, r, contentPath, resolvedPath.Cid())
```